website/blog/2022-11-21-react-native-core-contributor-summit-2022.md

```diff
@@ -0,0 +1,123 @@
+
+### ​​React Native Monorepo
+
+Releasing a new version of React Native is not trivial today. React Native is one of the most downloaded packages on NPM, and we want to make sure that our release process is smooth.
+
+That’s why we want to refactor the `react-native` repository and implement the **Monorepo RFC** ([#480](https://github.com/react-native-community/discussions-and-proposals/pull/480)).
+
+In this session, we initially brainstormed and collected opinions from every contributor, as it’s crucial that we evolve our repository to reduce the breaking changes for our downstream dependencies.
+
+We then started working on two fronts. First, we had to expand our Continuous Integration infrastructure to support our monorepo, adding [Verdaccio](https://verdaccio.org/) to our testing infrastructure. We then started renaming & adding scopes to several of our packages, resulting in 6 distinct contributions.
+
+You can track the status of this effort under this [umbrella issue](https://github.com/facebook/react-native/issues/34692) and we hope to share more on this effort in the near future.
+
+### Metro Web and Ecosystem Alignment
+
+[Metro](https://github.com/facebook/metro), our JavaScript Bundler, is a foundational and integrated part of the React Native development experience and we want to make sure it works with the latest standards in the JS ecosystem.
+
+The focus of this session was to discuss improving Metro's feature set to work better for web use cases and with the npm and bundler ecosystem. Two major areas of discussion:
+
+1. Adopting the `”exports”` ([package entry points](https://nodejs.org/api/packages.html#package-entry-points)) specification
+
+From the [Node.js documentation](https://nodejs.org/api/packages.html#package-entry-points):
+
+<!-- alex ignore clearly -->
+
+> The ["exports"](https://nodejs.org/api/packages.html#exports) provides a modern alternative to ["main"](https://nodejs.org/api/packages.html#main) allowing multiple entry points to be defined, conditional entry resolution support between environments, and **preventing any other entry points besides those defined in ["exports"](https://nodejs.org/api/packages.html#exports)**. This encapsulation allows module authors to define the public interface for their package clearly.\_
+
+Adopting the `“exports”` specification has a lot of potential. In this session, we debated on how to handle [Platform Specific Code](https://reactnative.dev/docs/platform-specific-code#platform-specific-extensions) with `“exports”`. Considering many factors, we came up with a fairly non-breaking rollout plan for `“exports”`, by adding a `“strict”` and `“non-strict”` mode to Metro resolver. We discussed how leveraging [builder-bob](https://github.com/callstack/react-native-builder-bob) would help library creators adopt the strict mode without friction.
```

Comment:
```suggestion
Adopting the `"exports"` specification has a lot of potential. In this session, we debated on how to handle [Platform Specific Code](/docs/platform-specific-code#platform-specific-extensions) with `"exports"`. Considering many factors, we came up with a fairly non-breaking rollout plan for `"exports"`, by adding a `"strict"` and `"non-strict"` mode to Metro resolver. We discussed how leveraging [builder-bob](https://github.com/callstack/react-native-builder-bob) would help library creators adopt the strict mode without friction.
```

We should not use the stylistic quotes in string code examples. Additionally this suggestion correct the internal URL.

Reply:
Thanks for doing that. Blaming the GDoc export feature :)